src/duckling/migration/README.md

```diff
@@ -0,0 +1,97 @@
+# Migration Engine
+
+Your duckling project has a `project/version.json` file that describes the project's version, expected editor version, and migrations.
+
+```
+{
+    projectVersion : "3.0",
+    editorVersion : "0.1",
+    migrations : [
+        {
+            "type" : "code", // This indicates the migration is custom code written for the project.
+            "updateTo" : "2.0", // Run the migration when updating maps to project version "2.0"
+            "name" : "migrations/make-collision100" // The migration script is {ProjectRoot}/project/migrations/make-collision100.js
+        },
+        {
+            "type" : "code",
+            "updateTo" : "3.0",
+            "name" : "migrations/add-drawable"
+        }
+    ]
+}
+```
+
+# Entity Migration
+
+Entity migrations are functions that accept a pre-migrated entity and returns the migrated entity. It can be used to add, remove, and modify attributes. Here is an example migration that adds a drawable attribute to every entity that does not have a drawable with a topDrawable. In the sample project above this would be in `project/migrations/add-drawable.js`
```

Comment:
perform some kind of operation on each entity in a map file. The function accepts a pre-migrated entity, and returns a migrated version of that same entity.